runs/run_taylor_fwd_inv.py

```diff
@@ -0,0 +1,99 @@
+# For fenics_ice copyright information see ACKNOWLEDGEMENTS in the fenics_ice
+# root directory
+
+# This file is part of fenics_ice.
+#
+# fenics_ice is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# fenics_ice is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with tlm_adjoint.  If not, see <https://www.gnu.org/licenses/>.
+
+# This script does not output anything. It is only use for
+# running taylor test verification on the forward runs outside
+# of the pytest environment
+
+import os
+os.environ["OMP_NUM_THREADS"] = "1"
+os.environ["OPENBLAS_NUM_THREADS"] = "1"
+
+import sys
+
+from runs import run_forward
+
+from fenics_ice.backend import clear_caches, compute_gradient, \
+    reset_manager, stop_manager, taylor_test
+
+def EQReset():
+    """Take care of tlm_adjoint EquationManager"""
+    # This prevents checkpointing errors when these run phases
+    # are tested after the stuff in test_model.py
+    reset_manager("memory")
+    clear_caches()
+    stop_manager()
+
+def test_tv_run_forward(config_file):
+
+    EQReset()
+
+    mdl_out = run_forward.run_forward(config_file)
+
+    slvr = mdl_out.solvers[0]
+
+    qoi_func = slvr.get_qoi_func()
+    cntrl = slvr.get_control()
+
+    slvr.reset_ts_zero()
+    J = slvr.timestep(adjoint_flag=1, qoi_func=qoi_func)[0]
+    dJ = compute_gradient(J, cntrl)
+
+    def forward_ts(cntrl, cntrl_init, name):
+        slvr.reset_ts_zero()
+        if (name == 'alpha'):
+            slvr._alpha = cntrl

```

Comment:
`slvr.set_control_fns([cntrl, slvr.beta], initial=True)`

Reply:
Hi @jrmaddison if I add that line after `slvr._alpha = cntrl ` I get the error below.
Should I add that instead of `slvr._alpha = cntrl ` ? 

```
Traceback (most recent call last):
  File "/home/brecinos/fenics_ice/runs//run_taylor_fwd_inv.py", line 105, in <module>
    test_tv_run_forward(sys.argv[1])
  File "/home/brecinos/fenics_ice/runs//run_taylor_fwd_inv.py", line 88, in test_tv_run_forward
    min_order = taylor_test(lambda cntrl_val: forward_ts(cntrl_val,
  File "/home/brecinos/tlm_adjoint/tlm_adjoint/verification.py", line 86, in taylor_test
    return taylor_test(forward, [M], J_val, dJ=dJ, ddJ=ddJ, seed=seed,
  File "/home/brecinos/tlm_adjoint/tlm_adjoint/verification.py", line 137, in taylor_test
    J_vals[i] = forward(*M1).value()
  File "/home/brecinos/fenics_ice/runs//run_taylor_fwd_inv.py", line 88, in <lambda>
    min_order = taylor_test(lambda cntrl_val: forward_ts(cntrl_val,
  File "/home/brecinos/fenics_ice/runs//run_taylor_fwd_inv.py", line 67, in forward_ts
    result = slvr.timestep(adjoint_flag=1, qoi_func=slvr.get_qoi_func())[1]
  File "/home/brecinos/fenics_ice/fenics_ice/solver.py", line 653, in timestep
    self.solve_mom_eq()
  File "/home/brecinos/fenics_ice/fenics_ice/solver.py", line 495, in solve_mom_eq
    momsolver.solve(annotate=annotate_flag)
  File "/home/brecinos/tlm_adjoint/tlm_adjoint/equations.py", line 507, in solve
    self.forward(self.X())
  File "/home/brecinos/tlm_adjoint/tlm_adjoint/equations.py", line 527, in forward
    self.forward_solve(X[0] if len(X) == 1 else X, deps=deps)
  File "/home/brecinos/fenics_ice/fenics_ice/solver.py", line 1516, in forward_solve
    solve(lhs == rhs, x, self._bcs, J=J,
  File "/home/brecinos/tlm_adjoint/tlm_adjoint/fenics/backend_overrides.py", line 254, in solve
    return backend_solve(*args, **kwargs)
  File "/home/brecinos/miniconda3/envs/fenics_upgrade/lib/python3.8/site-packages/dolfin/fem/solving.py", line 220, in solve
    _solve_varproblem(*args, **kwargs)
  File "/home/brecinos/miniconda3/envs/fenics_upgrade/lib/python3.8/site-packages/dolfin/fem/solving.py", line 266, in _solve_varproblem
    solver.solve()
RuntimeError:

*** -------------------------------------------------------------------------
*** DOLFIN encountered an error. If you are not able to resolve this issue
*** using the information listed below, you can ask for help at
***
***     fenics-support@googlegroups.com
***
*** Remember to include the error message listed below and, if possible,
*** include a *minimal* running example to reproduce the error.
***
*** -------------------------------------------------------------------------
*** Error:   Unable to solve nonlinear system with NewtonSolver.
*** Reason:  Newton solver did not converge because maximum number of iterations reached.
*** Where:   This error was encountered inside NewtonSolver.cpp.
*** Process: 0
***
*** DOLFIN version: 2019.1.0
*** Git changeset:  7e99df3564aea9dc05729ddb64498ae47b6bc15d
*** -------------------------------------------------------------------------
```